[ADD][16.0] report_footer_html

Shide opened the pull request:
Display HTML content in footer of chosen reports.
Compatible with all external reports of Odoo.

You can upload an small image or a large banner or just add some other HTML.

You must adjust Botton margin of the report. It's recommended to change Paper Format to a new one that has enough Bottom Margin to display the HTML of the report.

# Example:
Default format paper: A4 (Bottom Margin 20)
<img width="987" height="827" alt="image" src="https://github.com/user-attachments/assets/726d33f1-c07d-4ad7-be03-ed015fcf5bfc" />

## Picking Operations report without footer html and using default paper format:
<img width="1331" height="424" alt="image" src="https://github.com/user-attachments/assets/2d657bdd-a5dd-4315-a3d3-130fd0881938" />

<img width="793" height="131" alt="image" src="https://github.com/user-attachments/assets/b88fe05c-f184-4715-85a1-4f31ef1ac039" />

## Delivery Slip report with footer and using custom paper format (Bottom Margin 40)
<img width="1333" height="457" alt="image" src="https://github.com/user-attachments/assets/d216973e-56dc-4ad1-b075-bfc4cb656309" />

<img width="793" height="189" alt="image" src="https://github.com/user-attachments/assets/571cc95b-2bc2-463c-8013-8a2f66bf1aec" />



---

MT-11608 @moduon @rafaelbn @Gelojr @chienandalu @fcvalgar @yajo please review if you want 😄

Shide committed:
[ADD] report_footer_html

chienandalu commented on report_display_image_in_footer/models/res_company.py:
```diff
@@ -0,0 +1,15 @@
+    _inherit = "res.company"
+
+    footer_image = fields.Binary(
+        string="Footer Image",
```
For pre-commit complaints
```suggestion
```

yajo commented on report_display_image_in_footer/models/base_document_layout.py:
```diff
@@ -0,0 +1,20 @@
+    def _onchange_company_id(self):
+        super()._onchange_company_id()
+        for wizard in self:
+            wizard.footer_image = wizard.company_id.footer_image
```
issue: this code looks unnecessary, because `footer_image` is already a related field.
```suggestion
```

yajo commented on report_display_image_in_footer/readme/USAGE.md:
```diff
@@ -0,0 +1,4 @@
+To use this module, you need to:
+
+1. Go to Configure Document Layout wizard on Settings and upload an image on `Footer Image` field.
+1. Go to Settings / Technical / Paper Format and adjust `Bottom Margin (mm)` accordingly.
```
issue: use code formatting only for code or technical terms:
```suggestion
1. Go to Configure Document Layout wizard on Settings and upload an image on *Footer Image* field.
1. Go to Settings / Technical / Paper Format and adjust *Bottom Margin (mm)* accordingly.
```

yajo reviewed (approved):
Minor suggestions.

As a major one, did you think about adding an HTML field for the footer instead? You could put an image within, or anything else.

yajo reviewed (changes requested):
Oh, sorry I forgot something. Could you please rename the module to something shorter? Example: `report_footer_image` (or `report_footer_custom` or `report_footer_html` in case you opt for the suggestion in https://github.com/OCA/reporting-engine/pull/1077#pullrequestreview-3321706315).

fcvalgar reviewed (approved):
LGTM, 

Thank you so much @Shide 

<img width="3396" height="1211" alt="image" src="https://github.com/user-attachments/assets/4d215b0a-b7b9-44b7-a9db-7c61b91c16fe" />

Shide commented:
@yajo Your suggestion was incredible! Thanks!
I've changed the image to an HTML field and works perfectly.

Also reviewed your comments and changed readme and module name.

Thanks ❤️

rafaelbn reviewed (approved):
Tested in production 👍🏼

chienandalu commented:
> @yajo Your suggestion was incredible! Thanks! I've changed the image to an HTML field and works perfectly.

But the regular footer is already an HTML field, am I missing something else now?

Shide commented:
**This module will be changed soon.**

chienandalu reviewed (approved):
Nice 👍

yajo commented on report_footer_html/views/ir_actions_report_view.xml:
```diff
@@ -0,0 +1,15 @@
+        <field name="inherit_id" ref="base.act_report_xml_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//page[@name='advanced']/group" position="inside">
+                <field name="footer_html" widget="html" options='{"safe": True}' />
```
issue: it should be visible only in html-based report types:

<img width="1390" height="470" alt="Image" src="https://github.com/user-attachments/assets/648a6ce9-3322-4caa-b138-fa9c4465f862" />

Shide replied:
Done

yajo reviewed (changes requested):
Thanks! Much better. It works fine, functionally, although I found one small bug.

fcvalgar reviewed (approved):
LGTM, thank you so much @Shide 

<img width="3423" height="1218" alt="image" src="https://github.com/user-attachments/assets/ae21238e-3699-4981-bfc5-171cd11b0d87" />

<img width="3432" height="1217" alt="image" src="https://github.com/user-attachments/assets/014ae23f-caa3-40ce-886b-3b1c530fe17e" />

ArantxaSudon reviewed (approved):
Revisado funcionalmente
Muchas gracias @Shide 

<img width="2037" height="765" alt="imagen" src="https://github.com/user-attachments/assets/ec16a464-e76e-4bec-84c4-1ee1fd095633" />

yajo reviewed (approved):
/ocabot merge nobump
